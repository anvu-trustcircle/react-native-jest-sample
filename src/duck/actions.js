const fetch = require('isomorphic-fetch')

export const anAction = () => ({ type: 'AN_ACTION' })

export const success = () => {
  return {
    type: 'FETCH_DATA_SUCCESS'
  }
}

export const fetchData = () => {
  return dispatch => {
    return fetch('https://facebook.github.io/react-native/movies.json') // Some async action with promise
      .then(() => {
        dispatch(success())
      })
  }
}
