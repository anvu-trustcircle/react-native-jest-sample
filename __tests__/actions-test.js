import configureStore from 'redux-mock-store'
import * as action from '../src/duck/actions'
import thunk from 'redux-thunk'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

// Initialize mockstore with empty state
const initialState = {}
const store = mockStore(initialState)

// Run before test
beforeEach(() => {
  store.clearActions()
})

// Run after test
afterEach(() => {
  expect(store.getActions()).toMatchSnapshot()
})

it('should dispatch action', () => {
  // Dispatch the action
  store.dispatch(action.anAction())

  // Test if your store dispatched the expected actions
  const storeActions = store.getActions()
  const expectedPayload = { type: 'AN_ACTION' }
  expect(storeActions).toEqual([expectedPayload])
})

it('should execute fetch data', () => {
  // Return the promise
  return store.dispatch(action.fetchData()).then(() => {
    const storeActions = store.getActions()
    expect(storeActions[0]).toEqual({ type: 'FETCH_DATA_SUCCESS' })
  })
})
