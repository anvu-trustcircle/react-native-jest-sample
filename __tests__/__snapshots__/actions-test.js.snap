// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should dispatch action 1`] = `
Array [
  Object {
    "type": "AN_ACTION",
  },
]
`;

exports[`should execute fetch data 1`] = `
Array [
  Object {
    "type": "FETCH_DATA_SUCCESS",
  },
]
`;
